### DISCUSSION TITLE
Pico w Support for 16MB?

### DISCUSSION THREAD

**Discussion by FlydGitHub** [micropython/micropython#10883] [RP2040 / Pico]
Hi,

Can anyone point me to a source of: 

a Pico w with 16MB of off-chip Flash memory?

Thanks

**Comment by scruss** [comment]
None yet that I know of. Raspberry Pi seem to be only producing 2 MB Pico Ws, and I haven't seen any clone boards of the W.

**Comment by FlydGitHub** [comment]
Thank you Stewart,

Do you know how I can find out how much flash is avable in both MicroPython and CircuitPython?

Flyd

**Reply by scruss** [reply]
In MicroPython, the number of bytes free is given by

    os.statvfs('/')[3] * os.statvfs('/')[0]

More details in [os.statvfs()](https://docs.micropython.org/en/v1.19.1/library/os.html#os.statvfs). My Pico had about 1.3 MB free out of the 2 MB flash size. Pico Ws will have less free space

I think CircuitPython may be the same

**Comment by redhead-p** [comment]
The Arduino Nano RP2040 Connect has 16Mbytes of Flash.  There's a standard MicroPython port for this. It doesn't have the same radio as the Pico W and has other peripherals.  My understanding is that the network API should be the same. The board has the standard Arduino Nano format.

See [https://store.arduino.cc/products/arduino-nano-rp2040-connect-with-headers](url)
